src/Controller/ProvideAvailableShippingMethodsAction.php

```diff
@@ -31,6 +32,11 @@ public function __invoke(Request $request): Response
         $order = $this->orderPreparator->prepareFromRequest($request);
         $shipment = $order->getShipments()->get((int) $request->attributes->get('shipmentNumber'));
 
-        return new JsonResponse($this->availableShippingMethodsListProvider->__invoke($shipment));
+        $jsonResponse = [];
+        if ($shipment instanceof ShipmentInterface) {
+            $jsonResponse = $this->availableShippingMethodsListProvider->__invoke($shipment);
+        }
+
+        return new JsonResponse($jsonResponse, 200);
```

Comment:
What would you say for

```php
if ($shipment === null) {
    return new JsonResponse([]);
}

return new JsonResponse($this->availableShippingMethodsListProvider->__invoke($shipment));
```

we get rid of unneeded variable, and `200` code is default :)

Reply:
Sure, I can change that. Just looks redundant to me.